chutney/environment/src/test/java/com/chutneytesting/environment/api/environment/HttpEnvironmentApiTest.java

```diff
@@ -156,12 +156,11 @@ public void createEnvironment_returns_409_when_already_existing() throws Excepti
     }
 
     @Test
-    public void deleteEnvironment_deletes_it_from_repo() throws Exception {
-        mockMvc.perform(delete(environmentBasePath + "/env test"))
+    public void deleteEnvironment_does_not_deletes_last_env() throws Exception {
+        addAvailableEnvironment("env_test");
+        mockMvc.perform(delete(environmentBasePath + "/env_test"))
             .andDo(MockMvcResultHandlers.log())
-            .andExpect(status().isOk());
-
-        verify(environmentRepository, times(1)).delete(eq("env test"));
+            .andExpect(status().isBadRequest());
```

Comment:
Bad request seems odd because used when request syntax is not correct. I would choose 409 - Conflict i think.